qt/scientific_interfaces/Indirect/IndirectFittingModel.cpp

```diff
@@ -508,7 +510,8 @@ void IndirectFittingModel::addNewWorkspace(MatrixWorkspace_sptr workspace,
 }
 
 void IndirectFittingModel::removeWorkspace(std::size_t index) {
-  removeFittingData(index);
+  if (m_fittingData.size() > index)
```

Comment:
Should this if statement also check for non-negative index?

Reply:
The index will always be positive or zero because it is a `std::size_t`

Reply:
If one gives this function an invalid index, should the user be aware that this happened? Perhaps one should log a warning and/or throw an exception?